File: src/main/java/dev/shota/decompiler/window/utils/Style.java
package dev.shota.decompiler.window.utils;

import lombok.Getter;
import lombok.SneakyThrows;
import java.util.Base64;
import java.util.Objects;

public class Style {

    @Getter
    private final String data;

    @SneakyThrows
    public Style(String name) {
        byte[] css = Objects.requireNonNull(getClass().getClassLoader().getResourceAsStream(String.format("styles/%s.css", name))).readAllBytes();
        String base64 = Base64.getEncoder().encodeToString(css);
        data = String.format("data:text/css;base64,%s", base64);
    }

}

File: src/main/kotlin/net/pryoscode/decompiler/window/utils/Style.kt
package net.pryoscode.decompiler.window.utils

import com.github.sommeri.less4j.core.ThreadUnsafeLessCompiler
import net.pryoscode.decompiler.window.Window
import java.util.*

fun style(file: String): String {
    val less = String(Window.javaClass.classLoader.getResourceAsStream("styles/$file").readAllBytes())
    val css = ThreadUnsafeLessCompiler().compile(less).css
    val base64 = Base64.getEncoder().encodeToString((css).toByteArray())
    return "data:text/css;base64,$base64"
}